tests/system/aiplatform/test_featurestore.py

```diff
@@ -15,24 +15,172 @@
 # limitations under the License.
 #
 
+import logging
+
 from google.cloud import aiplatform
 from tests.system.aiplatform import e2e_base
 
+_TEST_USERS_ENTITY_TYPE_GCS_SRC = (
+    "gs://cloud-samples-data-us-central1/vertex-ai/feature-store/datasets/users.avro"
+)
+_TEST_MOVIES_ENTITY_TYPE_GCS_SRC = (
+    "gs://cloud-samples-data-us-central1/vertex-ai/feature-store/datasets/movies.avro"
+)
+
+_TEST_FEATURESTORE_ID = "movie_prediction"
+_TEST_USER_ENTITY_TYPE_ID = "users"
+_TEST_MOVIE_ENTITY_TYPE_ID = "movies"
+
+_TEST_USER_AGE_FEATURE_ID = "age"
+_TEST_USER_GENDER_FEATURE_ID = "gender"
+_TEST_USER_LIKED_GENRES_FEATURE_ID = "liked_genres"
+
+_TEST_MOVIE_TITLE_FEATURE_ID = "title"
+_TEST_MOVIE_GENRES_FEATURE_ID = "genres"
+_TEST_MOVIE_AVERAGE_RATING_FEATURE_ID = "average_rating"
+
 
 class TestFeaturestore(e2e_base.TestEndToEnd):
 
-    _temp_prefix = "temp-vertex-sdk-e2e-feature-store-test"
+    _temp_prefix = "temp_vertex_sdk_e2e_featurestore_test"
+
+    def test_end_to_end(self, shared_state, caplog):
 
-    def test_create_and_get_featurestore(self, shared_state):
+        caplog.set_level(logging.INFO)
 
         aiplatform.init(
             project=e2e_base._PROJECT, location=e2e_base._LOCATION,
         )
 
-        shared_state["resources"] = []
+        base_list_featurestores = len(aiplatform.Featurestore.list())
+        base_list_searched_features = len(aiplatform.Feature.search())
+
+        # Featurestore
+        featurestore_id = self._make_display_name(key=_TEST_FEATURESTORE_ID).replace(
+            "-", "_"
+        )[:60]
+        featurestore = aiplatform.Featurestore.create(featurestore_id=featurestore_id)
+        shared_state["resources"] = [featurestore]
+
+        get_featurestore = aiplatform.Featurestore(
+            featurestore_name=featurestore.resource_name
+        )
+        assert featurestore.resource_name == get_featurestore.resource_name
 
         list_featurestores = aiplatform.Featurestore.list()
-        assert len(list_featurestores) >= 0
+        assert len(list_featurestores) - base_list_featurestores == 1
```

Comment:
I'm not sure what is the style guide with OSS, should we follow one assert per test function or use subtest?

Reply:
Double checking the integration test standard (we did plan to have simple end to end test, but I can see this getting complicated), will update upon confirmation.